### PR TITLE
Unit test - request/writer.go

### PR DESCRIPTION
**Pull request by nathannaveen**
### 1. Why is this pull request needed and what does it do?

- Implemented unit tests for request/writer.go

### CODE REVIEW

**Comment by nathannaveen** [comment]
@yongtang 👀 please!

File: request/writer_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
+package request
+
+import (
+	"github.com/coredns/coredns/plugin/test"
+	"github.com/miekg/dns"
+	"reflect"
+	"testing"
+)
+
+func TestNewScrubWriter(t *testing.T) {
```
Comment (chrisohaver) [inline request/writer_test.go:10]: This test is not needed.  `NewScrubWriter` is a one line function - too simple to warrant this.

File: request/writer_test.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	test.ResponseWriter
+	written []byte
+}
+
+func (r *inmemoryWriter) WriteMsg(m *dns.Msg) error {
+	r.written, _ = m.Pack()
+	return r.ResponseWriter.WriteMsg(m)
+}
+
+func TestScrubWriter_WriteMsg(t *testing.T) {
```
Comment (chrisohaver) [inline request/writer_test.go:67]: This doesn't appear to be testing anything that `WriteMsg` does.

**Review by chrisohaver — CHANGES_REQUESTED** [review]
I don't think these tests are helpful.

**Pull request closed without merging**